Assets/MixedRealityToolkit.Tests/PlayModeTests/NearInteractionTouchableTests.cs

```diff
@@ -339,6 +339,78 @@ public IEnumerator NearInteractionTouchableOverlapQuerySaturation()
+        public IEnumerator NearInteractionTouchableUnityUiButton()
+        {
+            UnityUiUtilities.EnsureInputModule();
+
+            var canvas = UnityUiUtilities.CreateCanvas(0.002f);
+
+            var button = UnityUiUtilities.CreateButton(Color.gray, Color.blue, Color.green);
+            button.transform.SetParent(canvas.transform, false);
+            var text = UnityUiUtilities.CreateText("test");
+            text.transform.SetParent(button.transform);
+
+            canvas.transform.position = objectPosition;
+
+            yield return new WaitForFixedUpdate();
+            yield return null;
+
+            yield return PlayModeTestUtilities.ShowHand(Handedness.Right, inputSim);
+
+            using (var catcher = new UnityButtonEventCatcher(button))
+            {
+                // Touch started and completed when entering and exiting
+                yield return PlayModeTestUtilities.MoveHandFromTo(initialHandPosition, objectPosition, numSteps, ArticulatedHandPose.GestureId.Open, Handedness.Right, inputSim);
+                Assert.AreEqual(0, catcher.Click);
+                yield return PlayModeTestUtilities.MoveHandFromTo(objectPosition, initialHandPosition, numSteps, ArticulatedHandPose.GestureId.Pinch, Handedness.Right, inputSim);
+                Assert.AreEqual(1, catcher.Click);
+            }
+
+            UnityEngine.Object.Destroy(canvas.gameObject);
+        }
+
+        /// <summary>
+        /// Test Unity UI toggle button
+        /// </summary>
+        /// <returns></returns>
+        [UnityTest]
+        public IEnumerator NearInteractionTouchableUnityUiToggle()
+        {
+            UnityUiUtilities.EnsureInputModule();
+
+            var canvas = UnityUiUtilities.CreateCanvas(0.002f);
+
+            var toggle = UnityUiUtilities.CreateToggle(Color.gray, Color.blue, Color.green);
+            toggle.transform.SetParent(canvas.transform, false);
+            var text = UnityUiUtilities.CreateText("test");
+            text.transform.SetParent(toggle.transform);
+
+            canvas.transform.position = objectPosition;
+
+            yield return new WaitForFixedUpdate();
+            yield return null;
+
+            yield return PlayModeTestUtilities.ShowHand(Handedness.Right, inputSim);
+
+            using (var catcher = new UnityToggleEventCatcher(toggle))
+            {
+                // Touch started and completed when entering and exiting
+                yield return PlayModeTestUtilities.MoveHandFromTo(initialHandPosition, objectPosition, numSteps, ArticulatedHandPose.GestureId.Open, Handedness.Right, inputSim);
+                Assert.IsFalse(catcher.Value);
+                Assert.AreEqual(0, catcher.Changed);
+                yield return PlayModeTestUtilities.MoveHandFromTo(objectPosition, initialHandPosition, numSteps, ArticulatedHandPose.GestureId.Pinch, Handedness.Right, inputSim);
+                Assert.IsTrue(catcher.Value);
+                Assert.AreEqual(1, catcher.Changed);
```

Comment:
Would it be possible to also try toggling again (to show that the reverse transition also works?)

Reply:
Added that now.